test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +89,6 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
```

Comment:
Consider still having the assertEquals but only when the command in context isMutating.